mne/time_frequency/tfr.py

```diff
@@ -1512,8 +1512,15 @@ def plot_joint(self, timefreqs=None, picks=None, baseline=None,
             fmax = freq + freq_half_range
 
             data = tfr.data
-
-            pos = find_layout(tfr.info).pos if layout is None else layout.pos
+            
+            if layout is None:
+                loaded_layout = find_layout(tfr.info)
+            
+            # only use position information for channels from layout 
+            # whose names appear as a substring in tfr.ch_names
+            idx = [ix for ix, ch_name in enumerate(loaded_layout.names) 
```

Comment:
The code should not contain comments. If it does, is a usual indication that this should either be a function with a descriptive name or that it should recoded with variable names that allow to understand what the code does at a glance.

Reply:
@massich don't you think you're a bit strict here? I tend to think comments are ipso facto good, and this is just two nested comprehensions after all.